### PR TITLE
Fix widget path resolver

### PR DESCRIPTION
**Pull request by rusq**
`formcommon.CreateUrl()` tried to os.Sta()t the path of `templates/%s/<something>` and of course it always failed.

* Now it forms the list of widgets, then resolves the path.
* use filepath.Join instead of literal "/"
* Added go.mod

### CODE REVIEW

File: widgets/widgets.go [hunk 1]
```diff
@@ -74,13 +77,14 @@ func BaseWidget(style, inputType string) *Widget {
 		formcommon.HIDDEN,
 		formcommon.IMAGE,
 		formcommon.MONTH:
-		urls = append(urls, formcommon.CreateUrl("templates/%s/input.html"))
+		urls = append(urls, filepath.Join(templatesPrefix, "input.html"))
```
Comment (kirves) [inline widgets/widgets.go]: Are you sure you should .Join with templatesPrefix here?

File: widgets/widgets.go [hunk 2]
```diff
@@ -74,13 +77,14 @@ func BaseWidget(style, inputType string) *Widget {
...
 	default:
-		urls = append(urls, formcommon.CreateUrl("templates/%s/input.html"))
+		urls = append(urls, filepath.Join(templatesPrefix, "input.html"))
```
Comment (kirves) [inline widgets/widgets.go]: Are you sure you should .Join with templatesPrefix here?

File: widgets/widgets.go [hunk 3]
```diff
@@ -74,13 +77,14 @@ func BaseWidget(style, inputType string) *Widget {
...
 	}
 	styledUrls := make([]string, len(urls))
```
Comment (kirves) [inline widgets/widgets.go]: Not needed based on your modification

File: widgets/widgets.go [hunk 4]
```diff
@@ -74,13 +77,14 @@ func BaseWidget(style, inputType string) *Widget {
...
+	// resolve paths
 	for i := range urls {
-		styledUrls[i] = fmt.Sprintf(urls[i], style)
+		urls[i] = formcommon.CreateUrl(filepath.Join(templatesPrefix, urls[i]))
 	}
 	templ := template.Must(template.ParseFiles(styledUrls...))
```
Comment (kirves) [inline widgets/widgets.go]: You should be using "urls" here instread of "styledUrls"

File: widgets/widgets.go (hunk 2 above)
Comment (rusq) [reply widgets/widgets.go]: Yeah, fixed. Thanks for noticing! (and the above too)

**Comment by rusq** [comment]
Hey @kirves , I noticed that travis tries to build this on Go 1.2 and shits the pants because `gimme` fails to fetch Go 1.2, do you mind if I update the travis.yml to use more up-to-date version, or do you have sentiments about Go 1.2?

**Pull request closed without merging**